cli/spec.py

```diff
@@ -16,30 +20,187 @@
 TRIM_PARAMS = ['default', 'required']
 
 
-def cfg_file_to_dict(file_path):
+class ValueArgument(object):
+    """
+    Default argument type for InfraRed Spec
+    Resolves "None" Types with priority defaults.
+    """
+    settings_dir = None
+    subcommand = None
+
+    def __init__(self, value=None, required=None):
+        self.value = value
+        self.arg_name = None
+        self.required = None
+
+    @classmethod
+    def get_app_attr(cls, attribute):
```

Comment:
These classmethod's  in the ValueArgument are over-responsibility for that class which is the additional  clg type. I'd suggest to move these helper methods to something like Arguments Helper. This is up to you.